File: app/src/main/java/com/cogu/spylook/model/entity/Grupo.java
package com.cogu.spylook.model.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity(tableName = "grupos")
@NoArgsConstructor
@AllArgsConstructor
public class Grupo {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private String nombre;
    public int id_creador;
}

File: app/src/main/java/com/cogu/spylook/model/entity/Grupo.kt
package com.cogu.spylook.model.entity

import androidx.room.Entity
import androidx.room.ForeignKey

@Entity(
    tableName = "grupos",
    foreignKeys = [
        ForeignKey(
            entity = Anotable::class,
            parentColumns = ["idAnotable"],
            childColumns = ["idCreador"], // Relación con Anotable
            onDelete = ForeignKey.CASCADE // Si se elimina un Anotable, sus grupos también se eliminan
        )
    ]
)
class Grupo(
    idAnotable: Int = 0,
    nombre: String,
    @JvmField
    var colorFoto : Int,
    @JvmField
    var idCreador: Int = 0,
): Anotable(idAnotable, nombre) {
    constructor() : this(0, "", 0)
}
